broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -186,10 +171,19 @@ protected void onActorCloseRequested() {
     closeFuture = new CompletableActorFuture<>();
 
     actor.run(
-        () ->
-            // allows to await current transition to avoid concurrent modifications and
-            // transitioning
-            currentTransitionFuture.onComplete((nothing, err) -> super.closeAsync()));
+        () -> {
+          LOG.debug("Closing ZeebePartition {}", context.getPartitionId());
+          closing = true;
+
+          removeListeners();
+          context.getComponentHealthMonitor().removeComponent(zeebePartitionHealth.getName());
+
+          final var inactiveTransitionFuture = transitionToInactive();
+
+          // allows to await current transition to avoid concurrent modifications and
+          // transitioning
+          inactiveTransitionFuture.onComplete((nothing, err) -> super.closeAsync());
```

Comment:
:thought_balloon: Here we can have a gap where we have transitioned to inactive but haven't closed the actor yet. During that gap we could have another transition (for example to leader) happening that was scheduled earlier and has already passed the `!closing` check. I _think_ this is bad and should not happen?

Reply:
This would not happen, as far as I understand. 
If the other transition has already passed `!closing` check, then inactive transition is enqueued after the other transition. In `PartitionTransitionImpl`, we already ensure that all transitions are executed sequentially. So inactive transition will wait until the other transition is completed.

Reply:
I see, that makes sense and I agree that this is the case :+1: